### PR TITLE
Use HUBActionHandlerWrapper as HUBActionHandler protocol

### PR DESCRIPTION
**Pull request by JohnSundell**
This change is part of the work for https://github.com/spotify/HubFramework/issues/158, which has the end goal of making `HUBViewController` publicly initializable. To do this, we need to remove all dependencies on concrete internal classes.

This is the first step, which makes `HUBViewController` take any `HUBActionHandler` as part of its initializer, rather than relying on the concrete `HUBActionHandlerWrapper` implementation. To make this happen, logic that was previously delegated to the view controller around chaining of async actions needed to be moved into the action handler wrapper itself.

The test change that was made in `HUBViewControllerTests` needed to make made to avoid implicit autorelease pools from taking over the lifecycle of objects.

### CODE REVIEW

**Comment by JohnSundell** [comment]
@spotify/objc-dev

File: sources/HUBActionHandlerWrapper.m [hunk 1]
```diff
@@ -114,23 +116,26 @@ - (BOOL)handleActionWithContext:(id<HUBActionContext>)context
...
 
-#pragma mark - HUBAsyncActionDelegate
+#pragma mark - HUBAsyncActionWrapperDelegate
 
-- (void)actionDidFinish:(id<HUBAsyncAction>)action
+- (void)actionDidFinish:(HUBAsyncActionWrapper *)action
+        withContext:(id<HUBActionContext>)context
         chainToActionWithIdentifier:(nullable HUBIdentifier *)nextActionIdentifier
-        customData:(nullable NSDictionary<NSString *, id> *)nextActionCustomData
+        customData:(nullable NSDictionary<NSString *,id> *)nextActionCustomData
```
Comment (krris) [inline sources/HUBActionHandlerWrapper.m]: Shouldn't be a space between `NSString` and `id`? 🔬

File: sources/HUBAsyncActionWrapper.h [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+ *  Notify an async action wrapper's delegate that it finished
+ *
+ *  @param action The action that finished
+ *  @param context The context that the action was performed using
+ *  @param nextActionIdentifier The identifier of any chained action to perform
+ *  @param nextActionCustomData Any custom data to pass to the chained action
+ */
+- (void)actionDidFinish:(HUBAsyncActionWrapper *)action
+            withContext:(id<HUBActionContext>)context
+        chainToActionWithIdentifier:(nullable HUBIdentifier *)nextActionIdentifier
```
Comment (krris) [inline sources/HUBAsyncActionWrapper.h]: Strange indentation but there's no way to fix this 😅

File: sources/HUBActionHandlerWrapper.m (hunk 1 above)
Comment (JohnSundell) [inline sources/HUBActionHandlerWrapper.m]: There should! Great catch 👍

**Pull request merged**